src/main/java/org/generationcp/middleware/api/brapi/VariableServiceBrapiImpl.java

```diff
@@ -0,0 +1,260 @@
+		final CVTerm scaleTerm = this.daoFactory.getCvTermDao().getById(scaleId);
+
+		// Only update the categorical values and min, max if variable is not used in study
+		if (!isVariableUsedInStudy) {
+
+			// Updating of scale name, categories, min, max is only applicable
+			// if the variable is not used in study.
+			scaleTerm.setName(variable.getScale().getScaleName());
+
+			final List<CVTermRelationship> relationships = this.daoFactory.getCvTermRelationshipDao().getBySubject(scaleTerm.getCvTermId());
+			final Optional<CVTermRelationship>
+				dataTypeRelationship = relationships.stream().filter(r -> r.getTypeId() == TermId.HAS_TYPE.getId()).findAny();
+			final DataType dataType = dataTypeRelationship.isPresent() ? DataType.getById(dataTypeRelationship.get().getObjectId()) : null;
+
+			// Update valid values if datatype is categorical, this only updates the label
+			if (dataType == DataType.CATEGORICAL_VARIABLE) {
+				final List<Integer> valueIds =
+					relationships.stream().filter(r -> r.getTypeId() == TermId.HAS_VALUE.getId()).map(CVTermRelationship::getObjectId)
+						.collect(Collectors.toList());
+				final Map<String, CVTerm> categoricalValuesMap = this.daoFactory.getCvTermDao().getByIds(valueIds).stream()
+					.collect(Collectors.toMap(CVTerm::getName, Function.identity()));
+
+				for (final CategoryDTO categoryDTO : variable.getScale().getValidValues().getCategories()) {
+					if (categoricalValuesMap.containsKey(categoryDTO.getValue())) {
+						final CVTerm categoricalValueTerm = categoricalValuesMap.get(categoryDTO.getValue());
+						categoricalValueTerm.setDefinition(categoryDTO.getLabel());
+						this.daoFactory.getCvTermDao().update(categoricalValueTerm);
+					}
+				}
+			}
+
+			// Update min and max if datatype is numerical
+			if (dataType == DataType.NUMERIC_VARIABLE) {
+				final int maxTermId = TermId.MAX_VALUE.getId();
+				final int minTermId = TermId.MIN_VALUE.getId();
+				final Integer minScale = variable.getScale().getValidValues().getMin();
+				final Integer maxScale = variable.getScale().getValidValues().getMax();
+				this.updateScaleMinMax(scaleId, minScale, minTermId);
+				this.updateScaleMinMax(scaleId, maxScale, maxTermId);
+			}
+
+			this.daoFactory.getCvTermDao().update(scaleTerm);
+
+		}
+
+	}
+
+	private void updateScaleMinMax(final Integer scaleId, final Integer scaleSize, final int termId) {
+		if (scaleSize != null) {
+			this.daoFactory.getCvTermPropertyDao().save(scaleId, termId, String.valueOf(scaleSize), 0);
+		} else {
+			final CVTermProperty property = this.daoFactory.getCvTermPropertyDao().getOneByCvTermAndType(scaleId, termId);
+			if (property != null) {
+				this.daoFactory.getCvTermPropertyDao().makeTransient(property);
+			}
+		}
+	}
+
+	private void addObservationVariableToStudy(final Integer studyDbId, final Integer observationVariableDbId,
+		final String observationVariableName) {
+		final Integer projectId =
```

Comment:
Rename projectId to plotDatasetId. 
I think we should keep a list of plotDatasetIds where the trait has been added (or already exists). 
That way, say we have 100 studyDbIds (instances) for the same trial, we don't query 100x for the plotDatasetId and 100x for its variables.